client/src/app/site/guards/permission.guard.ts

```diff
@@ -1,28 +1,42 @@
 import { Injectable } from '@angular/core';
-import { CanLoad, Route } from '@angular/router';
+import { CanLoad, Route, UrlSegment } from '@angular/router';
 
+import { AuthCheckService } from '../services/auth-check.service';
 import { OperatorService } from '../services/operator.service';
 
 @Injectable({
     providedIn: `root`
 })
 export class PermissionGuard implements CanLoad {
-    public constructor(private operator: OperatorService) {}
+    public constructor(private operator: OperatorService, private authCheck: AuthCheckService) {}
 
-    public async canLoad(route: Route): Promise<boolean> {
+    public async canLoad(route: Route, segments: UrlSegment[]): Promise<boolean> {
+        if (await this.isOnForbiddenRoute(segments)) {
+            this.authCheck.forwardToOnlyMeeting();
+            return false;
+        }
+        if (route.data) {
+            const isAuthorized = await this.authCheck.isAuthorized(route.data);
+            if (!isAuthorized) {
+                this.authCheck.handleForbiddenRoute(route.data, segments);
+            }
+            return isAuthorized;
+        }
         return true;
-        // await this.operator.ready;
-        // const meetingPermissions: Permission[] = route.data?.['meetingPermissions'];
-        // const omlPermissions: OML[] = route.data?.['omlPermissions'];
-        // console.log(`canLoad`, meetingPermissions, omlPermissions);
-        // if (meetingPermissions?.length) {
-        //     console.log(`operator canload:`, this.operator.hasPerms(...meetingPermissions));
-        //     return this.operator.hasPerms(...meetingPermissions);
-        // }
-        // if (omlPermissions?.length) {
-        //     console.log(`operator canLoad:`, this.operator.hasOrganizationPermissions(...omlPermissions));
-        //     return this.operator.hasOrganizationPermissions(...omlPermissions);
-        // }
-        // return true;
+    }
+
+    /**
+     * Returns false if the user is on the dashboard, when he actually shouldn't be allowed to see it
+     */
+    private async isOnForbiddenRoute(segments: UrlSegment[]): Promise<boolean> {
+        if (segments.length === 0) {
+            await this.operator.ready;
+            try {
+                return !this.operator.knowsMultipleMeetings;
+            } catch (e) {
+                return true;
+            }
+        }
+        return false;
     }
```

Comment:
Can the implementation of `isOnForbiddenRoute` in the `PermissionGuard` and the implementation of `hasStateMeaning` in the `AuthGuard` be unified?